File: record/record.rs
const NRECORDS: usize = 16_777_216;

#[allow(dead_code)]
struct Record {
    id: i32,
}

fn main() {
    let mut records = Vec::with_capacity(NRECORDS);
    
    for i in 0..NRECORDS {
        let r = Record { id: i as i32 };
        records.push(r);
    }
}

File: record/record.zig
const std = @import("std");
const allocator = std.heap.page_allocator;

const NRECORDS = 8388608;

const Record = struct {
    id: i32,
};

pub fn main() !void {
    const records = try allocator.alloc(Record, NRECORDS);
    defer allocator.free(records);

    for (0..NRECORDS) |i| {
        const r = Record{ .id = @intCast(i) };
        records[i] = r;
    }
}
